pybind_interface/decide/decide.cpp

```diff
@@ -34,10 +34,15 @@ enum Instructions { AVX512F = 0, AVX2 = 1, SSE4_1 = 2, BASIC = 3};
 
 int detect_instructions() {
   Instructions instr = BASIC;
-  int info[4];
 
+  // Check for ARM architecture, specifically for Apple M1 or similar
+  #if defined(__aarch64__) && defined(__APPLE__)
```

Comment:
Lines 39-42 can be simplifies as
```C++
#if !defined(__aarch64__) || !defined(__APPLE__)
```

Reply:
Thanks a lot! I have added this in fff819e.